Source/WebCore/Modules/cache/DOMCache.cpp

```diff
@@ -41,15 +41,15 @@
 namespace WebCore {
 using namespace WebCore::DOMCacheEngine;
 
-Ref<DOMCache> DOMCache::create(ScriptExecutionContext& context, String&& name, DOMCacheIdentifier identifier, Ref<CacheStorageConnection>&& connection)
+Ref<DOMCache> DOMCache::create(ScriptExecutionContext* context, String&& name, DOMCacheIdentifier identifier, Ref<CacheStorageConnection>&& connection)
```

Comment:
I think this should still be a `ScriptExecutionContext&`, because if the script context is null then the object should not be created. We should probably add a check for `ScriptExecutionContext` in `DOMCacheStorage` and throw exception if the context is null.

Reply:
Ok, we can also do that. I hesitated between the 2.

Reply:
I'll just point out that your approach is more painful (requires a bit of refactoring at call sites) and I don't think there are major benefits, it just "feels" cleaner. See `findCacheOrCreate()` for example, currently returns a `Ref<DOMCache>` so it is never expected to fail.

Reply:
Thanks, adding a null check seems the right thing to do (given that there is no guarantee context() will not return null in previous code)